sql/server_component/mysql_thd_kill_notification_imp.h

```diff
@@ -0,0 +1,15 @@
+#ifndef SQL_SERVER_COMPONENT_MYSQL_THD_KILL_NOTIFICATION_IMP_H
+#define SQL_SERVER_COMPONENT_MYSQL_THD_KILL_NOTIFICATION_IMP_H
+
+#include "mysql/components/service_implementation.h"
+#include "mysql/components/services/mysql_thd_kill_notification.h"
+
+class Mysql_thd_kill_notification_imp {
+ public:
+  static DEFINE_BOOL_METHOD(subscribe,
+                            (MYSQL_THD thd, kill_notification_fn fn));
+
+  static void notify(THD *thd, kill_notification_fn fn);
```

Comment:
Should this be private (or private + being friend with a class invoking it)?

Reply:
Done.